x/utils/httpQueries.go

```diff
@@ -0,0 +1,40 @@
+package utils
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+)
+
+//queryFromAPI is generic function that can query every api and endpoint
+func queryFromAPI(url string, endpoint string, ptr interface{}) error {
+	resp, err := http.Get(fmt.Sprintf("%s/%s", url, endpoint))
+	if err != nil {
+		return err
+	}
+
+	defer resp.Body.Close()
+
+	bz, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return err
+	}
+
+	if err := json.Unmarshal(bz, &ptr); err != nil {
+		log.Print(string(bz))
+		return err
+	}
+
+	return nil
+}
+
+// QueryCoinGecko can query endpoint from coingecko
+func QueryCoinGecko(endpoint string, ptr interface{}) error {
+	url := "https://api.coingecko.com/api/v3"
+	if err := queryFromAPI(url, endpoint, ptr); err != nil {
+		return err
+	}
+	return nil
+}
```

Comment:
I don't think this should be put inside the `x/utils` folder since it will only be used by the price-related methods. Instead, it should be placed inside the module that uses it. We should place only generic methods here that can be used by a variety of different modules.